src/libs/ReportPreviewActionUtils.ts

```diff
@@ -0,0 +1,114 @@
+    const isApprover = isApproverMember(policy, getCurrentUserAccountID());
+    const isProcessing = isProcessingReport(report);
+    const isApprovalEnabled = policy ? policy.approvalMode && policy.approvalMode !== CONST.POLICY.APPROVAL_MODE.OPTIONAL : false;
+    const hasViolations = hasAnyViolations(report.reportID, violations);
+    const reportTransactions = transactions ?? getReportTransactions(report?.reportID);
+    return isExpense && isApprover && isProcessing && isApprovalEnabled && !hasViolations && reportTransactions.length > 0;
+}
+
+function canPay(report: Report, violations: OnyxCollection<TransactionViolation[]>, policy?: Policy) {
+    const isReportPayer = isPayer(getSession(), report, false, policy);
+    const isExpense = isExpenseReport(report);
+    const isPaymentsEnabled = arePaymentsEnabled(policy);
+    const isProcessing = isProcessingReport(report);
+    const isApprovalEnabled = policy ? policy.approvalMode && policy.approvalMode !== CONST.POLICY.APPROVAL_MODE.OPTIONAL : false;
+    const isSubmittedWithoutApprovalsEnabled = !isApprovalEnabled && isProcessing;
+    const isApproved = isReportApproved({report}) || isSubmittedWithoutApprovalsEnabled;
+    const isClosed = isClosedReport(report);
+    const hasViolations = hasAnyViolations(report.reportID, violations);
+    const isInvoice = isInvoiceReport(report);
+    const isIOU = isIOUReport(report);
+
+    if (!isReportPayer) {
+        return false;
+    }
+    return (isExpense && isPaymentsEnabled && (isApproved || isClosed) && !hasViolations) || ((isInvoice || isIOU) && isProcessing);
+}
+
+function canExport(report: Report, violations: OnyxCollection<TransactionViolation[]>, policy?: Policy) {
+    const isExpense = isExpenseReport(report);
+    const isExporter = policy ? isPrefferedExporter(policy) : false;
+    const isApproved = isReportApproved({report});
+    const isReimbursed = isSettled(report);
+    const isClosed = isClosedReport(report);
+    const hasAccountingConnection = hasAccountingConnections(policy);
+    const connectedIntegration = getConnectedIntegration(policy);
+    const syncEnabled = hasIntegrationAutoSync(policy, connectedIntegration);
+    const hasViolations = hasAnyViolations(report.reportID, violations);
+    if (isExpense && isExporter && hasAccountingConnection && !syncEnabled && !hasViolations) {
+        return isApproved || isReimbursed || isClosed;
+    }
+    return false;
+}
+
+function canReview(report: Report, violations: OnyxCollection<TransactionViolation[]>, policy?: Policy) {
+    const hasViolations = hasAnyViolations(report.reportID, violations);
+    const isSubmitter = isCurrentUserSubmitter(report.reportID);
+    const isApprover = isApproverMember(policy, getCurrentUserAccountID());
+    const areWorkflowsEnabled = policy ? policy.areWorkflowsEnabled : false;
+    return hasViolations && (isSubmitter || isApprover) && areWorkflowsEnabled;
+}
+
+function getReportPreviewAction(
```

Comment:
@jnowakow FYI, there were already two different methods named `getReportPreviewAction` and you added a third one 🙈

Reply:
Fact, but this one this one does something different. The other two refers to `ReportAction` entity and this one refers to action that user can undertake from report preview and we followed naming convention established in design doc - https://docs.google.com/document/d/1JZLr_5gX348cMZXNr6HUwi4QscOcDg3IY368FMAu3H4/edit?tab=t.0#bookmark=id.n099p6o4vamj